include/experimental/__p1673_bits/conjugated.hpp

```diff
@@ -44,135 +44,92 @@
 #define LINALG_INCLUDE_EXPERIMENTAL___P1673_BITS_CONJUGATED_HPP_
 
 #include <experimental/mdspan>
-#include <complex>
 
 namespace std {
 namespace experimental {
 inline namespace __p1673_version_0 {
 namespace linalg {
 
-template<class T>
-class conjugated_scalar {
-public:
-  using value_type = T;
-
-  conjugated_scalar(const T& v) : val(v) {}
-
-  operator T() const { return conj(val); }
-
-  template<class T2>
-  T operator* (const T2 upd) const {
-    return conj(val) * upd;
-  }
-
-  template<class T2>
-  T operator+ (const T2 upd) const {
-    return conj(val) + upd;
-  }
-
-  template<class T2>
-  bool operator== (const T2 upd) const {
-    return conj(val) == upd;
-  }
-
-  template<class T2>
-  bool operator!= (const T2 upd) const {
-    return conj(val) != upd;
-  }
-
-private:
-  const T& val;
-};
-
-template<class T1, class T2>
-auto operator* (const T1 x, const conjugated_scalar<T2> y) {
-  using std::conj;
-  return x * T2(y);
-}
-
-template<class Accessor, class T>
+template<class Accessor>
 class accessor_conjugate {
 private:
-  using size_type = typename extents<>::size_type;
-public:
-  using element_type  = typename Accessor::element_type;
-  using pointer       = typename Accessor::pointer;
-  using reference     = typename Accessor::reference;
-  using offset_policy = typename Accessor::offset_policy;
-
-  accessor_conjugate() = default;
+  Accessor accessor_;
 
-  accessor_conjugate(Accessor a) : acc(a) {}
+  using accessor_value_type =
+    std::remove_cv_t<typename Accessor::element_type>;
+  static constexpr bool accessor_value_is_arith =
+    std::is_arithmetic_v<accessor_value_type>;
 
-  reference access(pointer p, size_type i) const noexcept {
-    return reference(acc.access(p, i));
-  }
-
-  typename offset_policy::pointer offset(pointer p, size_type i) const noexcept {
-    return acc.offset(p, i);
-  }
-
-  element_type* decay(pointer p) const noexcept {
-    return acc.decay(p);
-  }
-private:
-  Accessor acc;
-};
+public:
+  using reference =
+    std::conditional_t<accessor_value_is_arith,
+      typename Accessor::reference,
+      conjugated_scalar<typename Accessor::reference, accessor_value_type>>;
 
-template<class Accessor, class T>
-class accessor_conjugate<Accessor, std::complex<T>> {
 private:
-  using size_type = typename extents<>::size_type;
+  using pre_const_element_type =
+    std::conditional_t<accessor_value_is_arith,
+      typename Accessor::element_type,
+      typename reference::value_type>;
+  
 public:
-  // FIXME If BLAS functions want to strip off accessor_conjugate for
-  // optimization, they will need a way to work with th underlying
-  // Accessor (which may not be the default one).
-
-  using element_type  = typename Accessor::element_type;
+  using element_type  = std::add_const_t<pre_const_element_type>;
```

Comment:
So the idea here is that you don't allow people to modify the values through something like `auto conj_a = conjugated(a); conj_a(0) = 0;`?

Reply:
@youyu3 Yes, absolutely.  `conjugated` was already specified to return a read-only view.  I realized when going through the implementation that I could just `add_const_t` the element type.  I didn't need the "nonconst version of the nested accessor," because I'm not changing the nested accessor; I'm returning a new accessor whose element type is the const version of the nested accessor's element type.